mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageZipperTest.java

```diff
@@ -0,0 +1,104 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault;
+
+import static org.apache.james.mailbox.backup.ZipAssert.EntryChecks.hasName;
+import static org.apache.james.mailbox.backup.ZipAssert.assertThatZip;
+import static org.apache.james.vault.DeletedMessageFixture.CONTENT;
+import static org.apache.james.vault.DeletedMessageFixture.DELETED_MESSAGE;
+import static org.apache.james.vault.DeletedMessageFixture.DELETED_MESSAGE_2;
+import static org.apache.james.vault.DeletedMessageFixture.MESSAGE_ID;
+import static org.apache.james.vault.DeletedMessageFixture.MESSAGE_ID_2;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import org.apache.commons.compress.archivers.zip.ZipFile;
+import org.apache.commons.compress.utils.SeekableInMemoryByteChannel;
+import org.apache.james.mailbox.backup.MessageIdExtraField;
+import org.apache.james.mailbox.backup.SizeExtraField;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DeletedMessageZipperTest {
+
+    private DeletedMessageZipper zipper;
+
+    @BeforeEach
+    void beforeEach() {
+        zipper = new DeletedMessageZipper();
+    }
+
+    @Test
+    void zipShouldPutEntriesToOutputStream() throws Exception {
+        ByteArrayOutputStream messageContents = zip(Stream.of(DELETED_MESSAGE, DELETED_MESSAGE_2));
+        try (ZipFile zipFile = zipFile(messageContents)) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MESSAGE_ID.serialize()).hasContent(CONTENT),
+                    hasName(MESSAGE_ID_2.serialize()).hasContent(CONTENT));
+        }
+    }
+
+    @Test
+    void zipShouldPutExtraFields() throws Exception {
+        ByteArrayOutputStream messageContents = zip(Stream.of(DELETED_MESSAGE));
+        try (ZipFile zipFile = zipFile(messageContents)) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MESSAGE_ID.serialize())
+                        .containsExtraFields(new MessageIdExtraField(MESSAGE_ID))
+                        .containsExtraFields(new SizeExtraField(CONTENT.length)));
+        }
+    }
+
+    @Test
+    void declareMultipleZippersCanDoZip() throws Exception {
```

Comment:
IMO 

```
@Test
void constructorShouldNotFailWhenCalledTwice() {
    assertThatCode(() -> new DeletedMessageZipper()).doesNotThrowAnyException();
}
```

is enough: we are just testing that the side effect hidden in this constructor is idempotent.